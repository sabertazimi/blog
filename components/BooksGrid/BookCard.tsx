'use client'

import Badge from '@/components/Badge'
import { Card, Meta } from '@/components/Card'
import { Book as BookIcon } from '@/components/Icons'
import { Span } from '@/components/Texts'
import { getColorByName } from '@/config'
import type { Book } from '@/types'

interface Props {
  book: Book
}

function BookCard({ book }: Props): JSX.Element {
  const { title, url, author, description } = book

  return (
    <Badge.Ribbon text={title} color={getColorByName(title)}>
      <Card
        title={(
          <a href={url}>
            <BookIcon className="align-top text-6xl" />
            <Span size="xl">{title}</Span>
          </a>
        )}
      >
        <Meta title={description} description={`Written by ${author}`} />
      </Card>
    </Badge.Ribbon>
  )
}

export default BookCard
