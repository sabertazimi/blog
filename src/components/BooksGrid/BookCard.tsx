import { BookOutlined } from '@ant-design/icons';
import { getColorByName } from '@config';
import type { Book } from '@types';
import { Badge, Card } from 'antd';
import React from 'react';

interface Props {
  data: Book;
}

const BookCard = ({ data }: Props): JSX.Element => {
  const { title, url, author, description } = data;

  return (
    <Badge.Ribbon text={title} color={getColorByName(title)}>
      <Card
        hoverable
        title={
          <a href={url}>
            <BookOutlined className="text-6xl align-top" />
            <span className="text-span-xl">{title}</span>
          </a>
        }
      >
        <Card.Meta title={description} description={`Written by ${author}`} />
      </Card>
    </Badge.Ribbon>
  );
};

export default BookCard;
