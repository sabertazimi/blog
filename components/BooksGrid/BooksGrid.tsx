import BookCard from './BookCard'
import Col from '@/components/Col'
import Row from '@/components/Row'
import { siteConfig } from '@/config'
import type { Book } from '@/types'

interface Props {
  books?: Book[]
}

function BooksGrid({ books = siteConfig.books }: Props): JSX.Element {
  return (
    <div role="grid">
      <Row>
        {books.map((book, index) => (
          <Col span={24} key={index}>
            <BookCard book={book} />
          </Col>
        ))}
      </Row>
    </div>
  )
}

export default BooksGrid
