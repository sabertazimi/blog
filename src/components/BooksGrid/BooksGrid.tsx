import type { Book } from '@types';
import { Col, Row } from 'antd';
import React from 'react';
import BookCard from './BookCard';

interface Props {
  bookList: Book[];
}

const BooksGrid = ({ bookList }: Props): JSX.Element => (
  <Row>
    {bookList.map((bookData, index) => (
      <Col span={24} key={index}>
        <BookCard data={bookData} />
      </Col>
    ))}
  </Row>
);

export default BooksGrid;
