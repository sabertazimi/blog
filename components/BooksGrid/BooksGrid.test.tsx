import { render } from '@utils'
import { axe } from 'jest-axe'
import BooksGrid from './BooksGrid'

describe('BooksGrid', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(<BooksGrid />)

    expect(container).toMatchSnapshot()
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(<BooksGrid />)

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })
})
