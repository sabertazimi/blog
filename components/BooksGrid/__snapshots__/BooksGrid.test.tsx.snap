// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksGrid should render correctly (snapshot) 1`] = `
<div>
  <div
    role="grid"
  >
    <div
      class="ant-row"
      role="row"
    >
      <div
        class="ant-col ant-col-24"
        role="cell"
      >
        <div
          class="ant-ribbon-wrapper"
        >
          <div
            class="ant-card ant-card-bordered card"
          >
            <div
              class="ant-card-head"
            >
              <div
                class="ant-card-head-wrapper"
              >
                <div
                  class="ant-card-head-title"
                >
                  <a
                    href="https://sabertazimi.github.io/awesome-notes"
                  >
                    <span
                      aria-label="book"
                      class="anticon anticon-book align-top text-6xl"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="book"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-260 72h96v209.9L621.5 312 572 347.4V136zm220 752H232V136h280v296.9c0 3.3 1 6.6 3 9.3a15.9 15.9 0 0022.3 3.7l83.8-59.9 81.4 59.4c2.7 2 6 3.1 9.4 3.1 8.8 0 16-7.2 16-16V136h64v752z"
                        />
                      </svg>
                    </span>
                    <span
                      class="ml-2 inline-block align-bottom text-5xl font-extrabold"
                    >
                      awesome-notes
                    </span>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="ant-card-body"
            >
              <div
                class="ant-card-meta"
              >
                <div
                  class="ant-card-meta-detail"
                >
                  <div
                    class="ant-card-meta-title"
                  >
                    Daily I Learned Notes
                  </div>
                  <div
                    class="ant-card-meta-description"
                  >
                    Written by sabertazimi
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="ant-ribbon ant-ribbon-placement-end"
            style="background: rgb(215, 51, 108);"
          >
            <span
              class="ant-ribbon-text"
            >
              awesome-notes
            </span>
            <div
              class="ant-ribbon-corner"
              style="color: rgb(215, 51, 108);"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
